pymagicc/api.py

```diff
@@ -39,38 +39,31 @@ def __init__(self, root_dir=None):
 
         if root_dir is not None:
             self.is_temp = False
-            if not exists(root_dir):
-                makedirs(root_dir)
         else:
             # Create a temp directory
             self.is_temp = True
             self.root_dir = mkdtemp(prefix="pymagicc-")
 
     def __enter__(self):
-        if not self.is_initialised():
-            self.init()
+        if self.is_temp and not exists(self.run_dir):
+            self.create_copy()
         return self
 
     def __exit__(self, *args, **kwargs):
-        self.close()
+        if self.is_temp:
```

Comment:
`remove_temp_copy` already checks `is_temp == True` before deletion. Perhaps L127 can be removed in preference to checking `is_temp` in `__exit__`. This will allow a user to explicitly remove a copied directory by calling `remove_temp_copy`, but temp directories are automatically cleaned up when using the context manager.

Reply:
Testing for `is_temp` twice I didn't like as well – I thought that it should be very difficult to accidentally remove a MAGICC package, so `pymagicc` should only remove temp copies.
Maybe the first check could go?

Reply:
That functionality would better align with the method name: `remove_temp_copy`

Reply:
Ok, then I'll remove the first check to make this clearer.